packages/devextreme/js/__internal/ui/list/m_list.edit.provider.ts

```diff
@@ -136,50 +155,65 @@ class EditProvider extends (Class.inherit({}) as new() => {}) {
     this._applyDecorators(DECORATOR_MODIFY_ELEMENT_METHOD, config);
   }
 
-  afterItemsRendered() {
+  afterItemsRendered(): void {
     this._applyDecorators(DECORATOR_AFTER_RENDER_METHOD);
   }
 
-  _prependBeforeBags($itemElement, config) {
-    const $beforeBags = this._collectDecoratorsMarkup(DECORATOR_BEFORE_BAG_CREATE_METHOD, config, LIST_ITEM_BEFORE_BAG_CLASS);
+  _prependBeforeBags($itemElement: dxElementWrapper, config: BagConfig): void {
```

Comment:
Since BagConfig doesn’t include $container from the start, this complicates its usage and forces it to be marked as optional. However, once the config is assembled, $container is always present at call time. It’s therefore better not to make any fields in BagConfig optional